contracts/contracts/poolBooster/PoolBoostCentralRegistry.sol

```diff
@@ -41,21 +47,45 @@ contract PoolBoostCentralRegistry is Governable {
      */
     function removeFactory(address _factoryAddress) external onlyGovernor {
         require(_factoryAddress != address(0), "Invalid address");
+        require(isApprovedFactory(_factoryAddress), "Not an approved factory");
 
-        uint256 length = factories.length;
-        bool factoryRemoved = false;
-        for (uint256 i = 0; i < length; i++) {
-            if (factories[i] != _factoryAddress) {
-                continue;
-            }
+        factories[_factoryAddress] = false;
+        emit FactoryRemoved(_factoryAddress);
+    }
 
-            factories[i] = factories[length - 1];
-            factories.pop();
-            emit FactoryRemoved(_factoryAddress);
-            factoryRemoved = true;
-            break;
-        }
-        require(factoryRemoved, "Not an approved factory");
+    /**
+     * @notice Emits a pool booster created event
+     * @dev    This has been created as a convenience method for the monitoring to have
+     *         an index of all of the created pool boosters by only listening to the
+     *         events of this contract.
+     * @param _poolBoosterAddress address of the pool booster created
+     * @param _ammPoolAddress address of the AMM pool forwarding yield to the pool booster
+     * @param _boosterType PoolBoosterType the type of the pool booster
+     */
+    function emitPoolBoosterCreated(
```

Comment:
We really should include the factory in the emit event. This way we know what kind of pool, and exactly the version that was created.

Reply:
Thanks great point done here: https://github.com/OriginProtocol/origin-dollar/pull/2400/commits/e45b7610f5b384c3b8dedf2bfea2f34883dec0b0